framework/src/main/java/org/checkerframework/common/basetype/messages.properties

```diff
@@ -80,6 +80,8 @@ purity.not.sideeffectfree.call=call to side-effecting %s not allowed in side-eff
 purity.more.deterministic=the method %s could be declared as @Deterministic
 purity.more.pure=the method %s could be declared as @Pure
 purity.more.sideeffectfree=the method %s could be declared as @SideEffectFree
+purity.incorrect.sideeffectsonly=the method side-effects %s
```

Comment:
@mernst I feel like this error message might be confusing/misleading in cases like
```java
@SideEffectsOnly("#1")
void foo(List<String> strs) {
    strs.add("...");
    baz();
}

baz() { ... }
```
The error message displayed will be:
```
baz()
    ^
the method side effects strs
```
Which might be true. Could we change this to "the method might side-effect ..."?

Reply:
Yes, that is a good change.  Thanks!